app/src/main/java/ch/epfl/skysync/screens/admin/UserManagement.kt

```diff
@@ -125,40 +128,53 @@ fun SearchBar(query: String, onQueryChanged: (String) -> Unit) {
 
 // Composable function to filter users by role.
 @Composable
-fun RoleFilter(onRoleSelected: (RoleType?) -> Unit, roles: List<RoleType>) {
+fun RoleFilter(onRoleSelected: (RoleType?) -> Unit, roles: List<RoleType>, count: Int) {
   var expanded by remember { mutableStateOf(false) } // State to manage dropdown expansion.
   var displayText by remember { mutableStateOf("Filter by role") }
+  Row(
+      modifier = Modifier.fillMaxWidth(),
+      verticalAlignment = Alignment.CenterVertically,
+  ) {
+    Column(modifier = Modifier.padding(16.dp)) {
+      Button(
+          onClick = { expanded = true },
+          colors =
+              ButtonDefaults.buttonColors(
+                  containerColor = lightOrange, contentColor = Color.Black)) {
+            Text(text = displayText)
+            Icon(
+                imageVector = Icons.Default.ArrowDropDown,
+                contentDescription = "Expand or collapse menu")
+          }
 
-  Column(modifier = Modifier.fillMaxWidth().padding(16.dp)) {
-    Button(
-        onClick = { expanded = true },
-        colors =
-            ButtonDefaults.buttonColors(containerColor = lightOrange, contentColor = Color.Black)) {
-          Text(text = displayText)
-          Icon(
-              imageVector = Icons.Default.ArrowDropDown,
-              contentDescription = "Expand or collapse menu")
-        }
-
-    // Dropdown menu for selecting roles.
-    DropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
-      DropdownMenuItem(
-          onClick = {
-            displayText = "Filter by role"
-            onRoleSelected(null)
-            expanded = false
-          },
-          text = { Text("All Roles") })
-      roles.forEach { role ->
+      // Dropdown menu for selecting roles.
+      DropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
         DropdownMenuItem(
             onClick = {
-              displayText = role.name
-              onRoleSelected(role)
+              displayText = "Filter by role"
+              onRoleSelected(null)
               expanded = false
             },
-            text = { Text(role.name) })

```

Comment:
use `role.description `instead of` role.name` to have a nicer string representation to display